docs/asdf/config.rst

```diff
@@ -40,7 +40,7 @@ the currently active config:
       all_array_compression: input
       all_array_compression_kwargs: None
       default_array_save_base: True
-      convert_unknown_ndarray_subclasses: True
+      convert_unknown_ndarray_subclasses: False
```

Comment:
Does this count as an API change? I'm not sure how asdf deals with changes like this and I don't know if there are any cases in practice that will be affected.

Reply:
I agree this counts as an API change.

When this feature was introduced (3.0.0) a warning was also added:
https://github.com/asdf-format/asdf/blob/abeef0ecfc756c09a42b7fbb954a0bcab399235d/asdf/yamlutil.py#L305-L310
and a section in the linked docs:
```
In a future version of asdf this default will change to False, a deprecation warning will be issued and finally the conversion of instances of subclasses will be removed.
```
We are now in the future! This PR adds the deprecation warning and asdf 5.0.0 we can remove the setting and the behavior.